polkadot-protocol/smart-contracts-basics/polkavm-design.md

```diff
@@ -0,0 +1,72 @@
+
+## PolkaVM
+
+[**PolkaVM**](https://github.com/paritytech/polkavm){target=\_blank} is a custom virtual machine optimized for performance with RISC-V-based architecture, supporting Solidity and additional high-performance languages. It serves as the core execution environment, integrated directly within the runtime. It features:
+
+- An efficient interpreter for immediate code execution
+- A planned JIT compiler for optimized performance
+- Dual-mode execution capability, allowing selection of the most appropriate backend for specific workloads
+- Optimized performance for short-running contract calls through the interpreter
+
+The interpreter remains particularly beneficial for contracts with minimal code execution, as it eliminates JIT compilation overhead and enables immediate code execution through lazy interpretation.
+
+### Architecture
+
+PolkaVM introduces two fundamental architectural differences compared to the Ethereum Virtual Machine (EVM):
+
+- **Register-Based design** - unlike EVM's stack-based architecture, PolkaVM utilizes a RISC-V register-based approach. This design:
+
+    - Employs a finite set of registers for argument passing instead of an infinite stack
+    - Facilitates efficient translation to underlying hardware architectures
+    - Optimizes register allocation through careful register count selection
+    - Enables simple 1:1 mapping to x86-64 instruction sets
+    - Reduces compilation complexity through strategic register limitation
+    - Improves overall execution performance through hardware-aligned design
+
+- **64-Bit word size**  - PolkaVM operates with a 64-bit word size, contrasting with EVM's 256-bit architecture:
+
+    - Enables direct hardware-supported arithmetic operations
+    - Maintains compatibility with Solidity's 256-bit operations through YUL translation
+    - Allows integration of performance-critical components written in lower-level languages
+    - Optimizes computation-intensive operations through native word size alignment
+    - Reduces overhead for operations not requiring extended precision
+    - Facilitates efficient integration with modern CPU architectures
+
+## Revive
+
+[**Revive**](https://github.com/paritytech/revive){target=\_blank} enables Solidity execution on PolkaVM through a particular compilation pipeline:
+
+- Utilizes the standard Solidity compiler (solc) as its foundation
+- Converts Solidity's [YUL](https://docs.soliditylang.org/en/latest/yul.html){target=\_blank} intermediate representation to RISC-V instructions
+- Maintains complete compatibility with all Solidity versions and features
+- Simplifies implementation by focusing on YUL transformation rather than full compiler development
+
+## Revive Remix
```

Comment:
I don't think Remix should be explained here as it is not part of the core protocol. This should be somewhere in develop